packages/valory/contracts/gnosis_safe_proxy_factory/contract.py

```diff
@@ -136,15 +136,18 @@ def build_tx_deploy_proxy_contract_with_nonce(  # pylint: disable=too-many-argum
         ):
             tx_parameters.update(ledger_api.try_get_gas_pricing())
 
-        if gas is not None:
-            tx_parameters["gas"] = Wei(gas)
+        tx_parameters["gas"] = (
+            Wei(gas) if gas != 0 else Wei(1)
```

Comment:
Wouldn't you get the same thing by leaving gas optional and checking for None instead of 0?

Reply:
Here yes, but below I need a gas value in the case where the gas estimate is not returned. And then the flow is much more transparent if the default is 0.